src/soundio/sounddeviceportaudio.cpp

```diff
@@ -407,12 +413,28 @@ bool SoundDevicePortAudio::isOpen() const {
     return m_pStream.load() != nullptr;
 }
 
+void SoundDevicePortAudio::finishedCallback() {
+    // This callback is called by PortAudio when the process callback returns
+    // paAbort to indicate the stream has to finish. This is triggered in the
+    // SoundDevicePortAudio::close() method.
+    std::unique_lock lock(m_finishedMutex);
+    m_bFinished = true;
+    m_finishedCV.notify_one();
+}
+
+void SoundDevicePortAudio::waitForFinished() {
+    // Wait until finishedCallback has been called. The timeout should never be
+    // reached, because when the process callback return paAbort, the stream
+    // will finish as soon as possible.
```

Comment:
// Wait until finishedCallback has been called. 
// The timeout is here as a last resort if the callback has stalled for some reasons. 

Correct?

Reply:
Yes, that is correct! (Never happened though during my stress test)